pkg/vector/entrypoint.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+set -eu
+
+export VECTOR_LOG="vector=info,vector::sources::util::unix_stream=warn"
+export VECTOR_LOG_FORMAT="text"
+export VECTOR_WATCH_CONFIG="true"
+export ALLOCATION_TRACING="true"
+
+DEFAULT_VECTOR_CONFIG="/etc/vector/vector.yaml"
+LIVE_CONFIG="/persist/vector/config/vector.yaml"
+CONFIG_CANDIDATE=${LIVE_CONFIG}.new
+PIDFILE=/var/run/vector.pid
+
+# --- Logging setup ---------------------------------------------------------
+# uncomment to write all logs to a file for debugging (e.g. if vector crashes)
+
+# # where to log everything
+# LOGFILE=/persist/vector/all.log
+# PIPE=/tmp/vector-logpipe
+
+# # ensure target dir exists
+# mkdir -p "$(dirname "$LOGFILE")"
+
+# # recreate pipe
+# [ -p "$PIPE" ] && rm "$PIPE"
+# mkfifo "$PIPE"
+
+# # start tee in background to write to file and stdout
+# tee -a "$LOGFILE" < "$PIPE" &
+# # redirect ALL stdout+stderr into pipe
+# exec > "$PIPE" 2>&1
+
+# # now all output from here on will be duplicated to console AND $LOGFILE
+
+# --- pre‐req check ---------------------------------------------------------
+
+if ! command -v inotifywait >/dev/null 2>&1; then
+  echo "ERROR: inotifywait not found. Install with:"
+  echo "  apk add --no-cache inotify-tools"
+  exit 1
+fi
+
+# --- initial setup --------------------------------------------------------
+
+if [ ! -f "$LIVE_CONFIG" ]; then
+  echo "No Vector config found at $LIVE_CONFIG"
+  echo "Copying default config from $DEFAULT_VECTOR_CONFIG"
+  mkdir -p "$(dirname "$LIVE_CONFIG")"
+  cp "$DEFAULT_VECTOR_CONFIG" "$LIVE_CONFIG"
+fi
+
+# --- launch & watch --------------------------------------------------------
+
+echo "Starting Vector with $LIVE_CONFIG"
+vector --config-yaml "$LIVE_CONFIG" &
+echo $! > "$PIDFILE"
+
+echo "Watching for new config at $CONFIG_CANDIDATE"
+inotifywait -m -e close_write "$(dirname "$CONFIG_CANDIDATE")" |
+while read _ _ changed; do
```

Comment:
wondering if it is possible to use 'vector' '--watch-config' option here.

Reply:
We are using this option for the main vector process https://github.com/lf-edge/eve/blob/79092d8258e311027e38cca86f4e6a26ac448a60/pkg/vector/entrypoint.sh#L6

`vector validate` however is a one-shot command and doesn't support this option